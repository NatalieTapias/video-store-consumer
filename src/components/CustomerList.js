import React from 'react';
import Customer from './Customer';
import PropTypes from 'prop-types';

const CustomerList = ({allCustomers, selectCustomerCallback}) => {
  
    return (
      <div className="row">
      {(allCustomers).map((oneCustomer, i) => {
        return(
          <Customer
            id={oneCustomer.id}
            name={oneCustomer.name}
            account_credit={oneCustomer.account_credit}
            movies_checked_out_count={oneCustomer.movies_checked_out_count}
            key={i}
            selectCustomerCallback={selectCustomerCallback}
            />);
          })}
      </div>
    );
  };
  
export default CustomerList;

CustomerList.propTypes = {
  allCustomers: PropTypes.array,
  selectCustomerCallback: PropTypes.func,
};