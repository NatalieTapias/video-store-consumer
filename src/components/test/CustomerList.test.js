import React from 'react';
import { render, cleanup } from '@testing-library/react'
import CustomerList from '../CustomerList';

describe('CustomerList', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <CustomerList 
      allCustomers={[]}
      selectCustomerCallback={() => {}}
      /> 
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});