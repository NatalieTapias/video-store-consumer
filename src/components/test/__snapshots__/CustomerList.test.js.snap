// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerList that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <h2
    class="title"
  >
    Rental Library
  </h2>
  <div
    class="row"
  />
</DocumentFragment>
`;
